sqlmesh/core/test/definition.py

```diff
@@ -95,10 +94,10 @@ def assert_equal(self, expected: pd.DataFrame, actual: pd.DataFrame) -> None:
         """Compare two DataFrames"""
         self._add_missing_columns(expected, actual)
 
-        # Two astypes are necessary, pandas converts strings to times as NS, but if the actual
-        # is US, it doesn't take affect until the 2nd try!
+        # Two astypes are necessary, pandas converts strings to times as NS,
+        # but if the actual is US, it doesn't take effect until the 2nd try!
         actual_types = actual.dtypes.to_dict()
-        expected = expected.astype(actual_types).astype(actual_types)
+        expected = expected.astype(actual_types, errors="ignore").astype(actual_types, errors="ignore")
```

Comment:
FYI: the `errors="ignore"` were added here (mostly) because of the following case: an expected output column is `None` when the actual column is non-`None`. I was getting this error (which is also true in main):

```
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
```

Reference for the `errors` kwarg: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.astype.html

Reply:
It seemed in general we might not wanna fail in the `astype` and report the related DataFrame error, but instead let it go to the df comparison stage and show the diff itself? Let me know if this assumption doesn't look right.